Analysis/Producer.py

```diff
@@ -0,0 +1,120 @@
+            self.df = self.df.Filter(f'tau{tau_idx}_idDeepTau{self.deepTauYear}{self.deepTauVersion}VSjet >= {Utilities.WorkingPointsTauVSjet.Medium.value}')
+        self.df = self.df.Filter('std::copysign(tau1_charge, tau2_charge)<0')
+        self.df = self.df.Filter('HLT_ditau')
+
+
+
+
+def createHistograms(df_dict, var):
+    hists = {}
+    x_bins = Utilities.ListToVector(plotter.hist_cfg[var]['x_bins'], "double")
+    model = ROOT.RDF.TH1DModel("", "", x_bins.size()-1, x_bins.data())
+    for sample in df_dict.keys():
+        weight_names=[]
+        for weight in weights_to_apply:
+            weight_names.append(weight if weight in df_dict[sample].GetColumnNames() else "1")
+            if weight == 'weight_Central' and weight in df_dict[sample].GetColumnNames():
+                weight_names.append("1000")
+        weight_str = " * ".join(weight_names)
+        print(weight_str)
+        df_dict[sample]=df_dict[sample].Define("weight",weight_str)
+        hists[sample] = df_dict[sample].Histo1D(model,var, "weight")
+    return hists
+
+
+
+if __name__ == "__main__":
+    import argparse
+    import PlotKit.Plotter as Plotter
+    import yaml
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--period', required=False, type=str, default = 'Run2_2018')
+    parser.add_argument('--version', required=False, type=str, default = 'v2_deepTau_v2p1')
+    parser.add_argument('--vars', required=False, type=str, default = 'tau1_pt')
+    parser.add_argument('--mass', required=False, type=int, default=500)
+    args = parser.parse_args()
+    sys.path.append(os.environ['ANALYSIS_PATH'])
+
+    abs_path = os.environ['CENTRAL_STORAGE']
+    anaTuplePath= "/eos/home-k/kandroso/cms-hh-bbtautau/anaTuples/Run2_2018/v2_deepTau_v2p1/" # os.path.join(os.environ['CENTRAL_STORAGE'], 'anaTuples', args.period, args.version)
+
+    page_cfg = "config/plot/cms_stacked.yaml"
+    page_cfg_custom = "config/plot/2018.yaml"
+    hist_cfg = "config/plot/histograms.yaml"
+    inputs_cfg = "config/plot/inputs.yaml"
+    with open(inputs_cfg, 'r') as f:
+        inputs_cfg_dict = yaml.safe_load(f)
+
+    plotter = Plotter.Plotter(page_cfg=page_cfg, page_cfg_custom=page_cfg_custom, hist_cfg=hist_cfg, inputs_cfg=inputs_cfg_dict)
+
+    dataframes = {}
+
+    for sample in files.keys():
+        rootFiles = [anaTuplePath+f + ".root" for f in files[sample]]
+        df = ROOT.RDataFrame("Events", rootFiles)
+        anaskimmer = AnaSkimmer(df)
+        anaskimmer.deepTauVersion = args.version.split('_')[-1]
+        if(sample in signals):
+            for input in inputs_cfg_dict:
+                name = input['name']
+                if(name == sample):
+                    input['title']+= f"mass {args.mass}"
+            anaskimmer.df = anaskimmer.df.Filter(f"X_mass=={args.mass}")
+        anaskimmer.skimAnatuple()
+        dataframes[sample] = anaskimmer.df
+
+    for var in args.vars.split(','):
```

Comment:
create histograms for all vars without calling `GetValue`. Then do another loop to plot.